File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [];
    for(let i = 0 ; i<n ; i++){
        let num1 = Number(Number(arr1[i]).toString(2)) ;
        let num2 = Number(Number(arr2[i]).toString(2)) ;
       // console.log(num1+num2)
        let str = check(n , String(num1+num2)); 
        answer.push(str); 
    }
    answer = answer.map(item => item.replaceAll("2" , "#").replaceAll("1" , "#").replaceAll("0" , " "));
    return answer;
}

function check(n , str){
    let zero = "0" ; 
    let i = n - str.length ; 
    zero = zero.repeat(i) ;
    return zero + str ; 
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*; 
class Solution {
    private int n ; 
    public String[] solution(int n, int[] arr1, int[] arr2) {
        this.n = n ; 
        String[] answer = new String[n] ; 
        for(int i = 0 ; i<arr1.length ; i++){
            String str = change(arr1[i] , arr2[i]);
            answer[i] = str.replaceAll("1" , "#").replaceAll("0" , " ") ;
        }
       
        
        return answer;
    }
    
    public String change(int num1 , int num2){
        StringBuilder sb1 = new StringBuilder() ; 
        StringBuilder sb2 = new StringBuilder() ; 
        while(true){
            if(sb1.length() == n && sb2.length() == n){
                break ; 
            }

            sb1.append(num1 % 2) ;
            sb2.append(num2 % 2) ; 
            num1 = num1 / 2 ;
            num2 = num2 / 2 ; 
 
        } 
        String[] str2 = sb2.reverse().toString().split(""); 
        String[] str1 = sb1.reverse().toString().split("");
   
        String[] arr = new String[sb2.length()] ; 
        Arrays.fill(arr , "1") ; 
        for(int i = 0 ; i<arr.length ; i++){
            if(str2[i].equals("0") && str1[i].equals("0")){
                arr[i] = "0" ; 
            }
        }        
        return String.join("" , arr) ;
        
    }
}